src/icmp/pinger.cc

```diff
@@ -233,14 +227,14 @@ main(int, char **)
             icmp4.Recv();
         }
 
-        if (PINGER_TIMEOUT + last_check_time < squid_curtime) {
+        if ((last_check_time + PINGER_TIMEOUT) < std::chrono::system_clock::now()) {
```

Comment:
TLDR: Please try to reuse Stopwatch instead. Let me know if I should attempt to do that first.

----

Long version:

I support migration to std::chrono, but I am against mixing std::chrono::system_clock::now() and old squid_curtime/etc. code unless absolutely necessary, especially in high-level code like this one. The code may continue to "work" on the surface, but having essentially two "current time" clocks is a serious complication that is best to avoid for basic engineering reasons.

To properly migrate to chrono, we should be upgrading existing Squid time APIs (including squid_curtime) to use chrono (including std::chrono::system_clock::now()), not replacing squid_curtime with system_clock::now() calls in high-level code like this one. If you want to work on that migration, let's discuss whether it can be structured to make this and similar changes unnecessary.

Alternatively, you may be able to migrate this particular code to use std::chrono without running into any of these complicated issues by reusing an existing chrono-based Squid class called Stopwatch! Please note that Stopwatch measures time _intervals_, not absolute time, so its API and its users do not conflict with the "one current time for all" principle. I recommend trying this alternative first. It also has the advantage of code reuse, of course.

Reply:
Done in [23b64fd](https://github.com/squid-cache/squid/pull/1277/commits/23b64fdccb2bcf68df6832cf7a5a54d404c9eca9)